src/libsyntax/ext/tt/quoted.rs

```diff
@@ -76,6 +76,8 @@ pub enum TokenTree {
     Delimited(Span, Rc<Delimited>),
     /// A kleene-style repetition sequence with a span
     Sequence(Span, Rc<SequenceRepetition>),
+    /// Matches a nonterminal. This is only used in the left hand side of MBE macros.
+    Match(Span, ast::Ident /* name to bind */, ast::Ident /* kind of nonterminal */),
```

Comment:
With a bit more refactoring I could also move `Nonterminal::SubstNt` here -- I think `MacroVariable` would be a better fit for that. `MacroVariableDeclaration` is too long though; how about `MetaVarDecl`?